Add IOCTL to disable frame boundary preemption

NishadSaraf opened the pull request:
Add IOCTL to disable frame boundary preemption.

maxzhen reviewed (changes requested):
It is confusing to have a "disable_frame_boundary_preempt_state" state. Can we just do "frame_boundary_preempt" state and it can be either disabled or enabled. It seems that we can have one data structure for enabling / disabling / fetching current state?

mamin506 commented:
retest this please

mamin506 commented:
@NishadSaraf , please fix checkpatch issues. 
Take a look at https://github.com/amd/xdna-driver?tab=readme-ov-file#checkpatch

NishadSaraf commented:
> It is confusing to have a "disable_frame_boundary_preempt_state" state. Can we just do "frame_boundary_preempt" state and it can be either disabled or enabled. It seems that we can have one data structure for enabling / disabling / fetching current state?


> @NishadSaraf , please fix checkpatch issues. Take a look at https://github.com/amd/xdna-driver?tab=readme-ov-file#checkpatch


Hi @maxzhen, @mamin506,
Thanks for the review. I have addressed these comments in the latest version.

mamin506 commented on src/driver/amdxdna/aie2_pci.c:
```diff
@@ -273,6 +291,17 @@ static int aie2_mgmt_fw_init(struct amdxdna_dev_hdl *ndev)
+		return ret;
+	}
+
+	/**
+	 * TODO: Fair scheduling feature is a Strix only feature. Cleanup the below implementation
+	 * once we have a generation specific management firmware initialization.
+	 */
```
" /** " is special for kernel-doc tool. It is a hint that this is part of the linux document. Only use /** when you know what you are doing.

 ```suggestion
	/*
	 * TODO: Fair scheduling feature is a Strix only feature. Cleanup the below implementation
	 * once we have a generation specific management firmware initialization.
	 */
```

mamin506 reviewed (changes requested):
Just one minor change is need. For the rest of the change, look good to me.

mamin506 commented:
retest this please
